lib/strings.nix

```diff
@@ -403,6 +403,15 @@ rec {
       name = builtins.head (splitString sep filename);
     in assert name !=  filename; name;
 
+  /* Get string of the first two parts (major and minor in "Semantic Versioning")
+     of a version.
+
+     Example:
+       majorMinorVersion "1.2.3"
+       => "1.2"
+  */
+  majorMinorVersion = v: builtins.concatStringsSep "." (lib.lists.take 2 (splitString "." v));
```

Comment:
How about `lib.semver.majorVersion` and `lib.semver.minorVersion` or is this too verbose?

Reply:
How about `lib.semver.major`, `lib.semver.minor`, `lib.semver.majorMinor`?

Reply:
Is the name `semver` common enough, that people will figure out the meaning on there own?

Reply:
But maybe this becomes clear when it is used in the code and the passed variable is called `version`.

Reply:
It is not limited to semver – see the GNOME example.

Reply:
It seems like "major" and "minor" are pretty universally the first and second parts of a version, so how about we make `lib.version.major`, `lib.version.minor`, and `lib.version.majorMinor`? We can also move `lib.strings.versionOlder`, and `lib.strings.versionAtLeast` into there as `lib.version.older` and lib.version.atLeast` later.